app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt

```diff
@@ -220,19 +222,28 @@ class TransactionListViewModel
 
     private fun Transaction.toCustomTransactionView(
         txInfo: TransactionInfo.Custom,
-        safeAddress: Solidity.Address,
+        safes: List<Safe>,
         awaitingYourConfirmation: Boolean,
         isConflict: Boolean
     ): TransactionView =
-        if (!isCompleted(txStatus)) queuedCustomTransaction(txInfo, safeAddress, awaitingYourConfirmation, isConflict)
-        else historicCustomTransaction(txInfo, safeAddress)
+        if (!isCompleted(txStatus)) queuedCustomTransaction(txInfo, safes, awaitingYourConfirmation, isConflict)
+        else historicCustomTransaction(txInfo, safes)
 
     private fun Transaction.historicCustomTransaction(
         txInfo: TransactionInfo.Custom,
-        safeAddress: Solidity.Address
+        safes: List<Safe>
     ): TransactionView.CustomTransaction {
-        //FIXME https://github.com/gnosis/safe-client-gateway/issues/189
-        val isIncoming: Boolean = txInfo.to == safeAddress
+
+        val address = txInfo.to.asEthereumAddressString()
+        val localName = safes.find { it.address == txInfo.to }?.localName
+        val toInfo = txInfo.toInfo
+
+        val addressInfo = when {
+            localName != null -> AddressInfoData.Local(localName, address)
+            toInfo != null -> AddressInfoData.Remote(toInfo.name, toInfo.logoUri, address)
+            else -> AddressInfoData.Default
+        }
```

Comment:
Could this (line 237 - 245) be extracted into a method to avoid duplication? Looks like this is the same code as in line 271 - 279.